### DISCUSSION TITLE
[#1040932] Open "script binding" to normal script api pack

### DISCUSSION THREAD

**Discussion by AbbottMc** [Mojang/minecraft-editor#102] [Feedback]
Noticed that you've using somthing like "binding" things in editor script to open api for editor pack. So is it possible to open this capability to the community in future development so that we can get rid of a bunch of duplicate code in different packs

**Comment by rlandav** [comment]
It's not something we are considering in the near term, particularly as we are fleshing out this pattern further. Simultaneously, we are also ensuring bundles are debuggable properly in BP's as well, so that mechanism of dependency sharing will remain possible. There are absolutely trade offs though. Individual bundles containing their own dependency versions allowing for isolated and specific versions of dependencies as needed, but at the cost of code bloat on a per pack basis which a common dependency module addresses.

We are aware of the advantages of this mechanism for potentially sharing dependencies, so we will be keeping this in mind for the future.

**Reply by AbbottMc** [reply]
Thank you very much for your reply! It's very excited that you guys think so too. Looking forward to seeing this feature implemented in the Bedrock Edition in the future!
